### PR TITLE
Introduce DownloadThumbInterceptor

### PR DESCRIPTION
**Pull request by VDostoyevskiy**

### CODE REVIEW

File: app/src/main/kotlin/com/hippo/ehviewer/spider/DownloadInfoMagics.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package com.hippo.ehviewer.spider
+
+import com.hippo.ehviewer.client.data.GalleryInfo
+import com.hippo.ehviewer.client.thumbUrl
+import com.hippo.ehviewer.dao.DownloadInfo
+
+object DownloadInfoMagics {
+    private const val DOWNLOAD_INFO_DIRNAME_URL_MAGIC = "Lolicon"
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/spider/DownloadInfoMagics.kt:8]: Serious?

File: app/src/main/kotlin/com/hippo/ehviewer/spider/DownloadInfoMagics.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    private const val DOWNLOAD_INFO_DIRNAME_URL_SEPARATOR = "~"
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/spider/DownloadInfoMagics.kt:9]: Don't use this as dirname may contain it.

Comment (FooIbar) [reply app/src/main/kotlin/com/hippo/ehviewer/spider/DownloadInfoMagics.kt:9]: I'd say use `|`, which is not allowed in both URLs and filenames.

Comment (VDostoyevskiy) [reply app/src/main/kotlin/com/hippo/ehviewer/spider/DownloadInfoMagics.kt:9]: SGTM

File: app/src/main/kotlin/com/hippo/ehviewer/spider/DownloadInfoMagics.kt (hunk 1 above)
Comment (FooIbar) [reply app/src/main/kotlin/com/hippo/ehviewer/spider/DownloadInfoMagics.kt:8]: Just use a regular one, like `$` or whatever.

**Review by FooIbar — APPROVED** [review]

**Pull request merged**